Add validate access token code sample for Java (#1163)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/validate-access-tokens/overview/java/accesstoken.md

See the instructions for the [Okta JWT Verifier for Java](https://github.com/okta/okta-jwt-verifier-java).

### Required Libraries

To validate a JWT, you will need the Okta JWT Verifier for Java library.

Include the lines below in your Maven `pom.xml`:

```xml
<dependency>
   <groupId>com.okta.jwt</groupId>
   <artifactId>okta-jwt-verifier</artifactId>
   <version>${okta-jwt.version}</version>
</dependency>
  
<dependency>
  <groupId>com.okta.jwt</groupId>
  <artifactId>okta-jwt-verifier-impl</artifactId>
  <version>${okta-jwt.version}</version>
  <scope>runtime</scope>
</dependency>
```

### Validate Access Token

Create the Okta JWT Verifier using the `JwtVerifiers` class:

```java
AccessTokenVerifier jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
    .setIssuer("https://{yourOktaDomain}/oauth2/default")
    .setAudience("api://default")                // defaults to 'api://default'
    .setConnectionTimeout(Duration.ofSeconds(1)) // defaults to 1s
    .setReadTimeout(Duration.ofSeconds(1))       // defaults to 1s
    .build();
```

> This helper class configures a JWT parser with the details found through the [OpenID Connect discovery endpoint](https://developer.okta.com/docs/reference/api/oidc/#well-known-openid-configuration).  The public keys used to validate the JWTs will also be retrieved and cached automatically.

Reviewer: ```suggestion
This helper class configures a JWT parser with the details found through the [OpenID Connect discovery endpoint](/docs/reference/api/oidc/#well-known-openid-configuration).  The public keys used to validate the JWTs will also be retrieved and cached automatically.
```

After you have a `JwtVerifier` from the above section and an `access_token` from a successful sign in, or from a Bearer token 

> in the authorization header, you will need to make sure that it is still valid. All you need to do is call the 

Reviewer: Let's remove this newline.

`decode` method (where `jwtString` is your access token in string format).

```java
Jwt jwt = jwtVerifier.decode(jwtString);
```

This will validate your JWT for the following:

> - Token expiration date

Reviewer: ```suggestion
- Token expiration time
```

> - the time it was issue at

Reviewer: ```suggestion
- the time it was issued at
```

> - The 'token not valid before' date

Reviewer: The time it was issued at

> - The token issuer matches the expected value passed into the above helper

Reviewer: ```suggestion
- that the token issuer matches the expected value passed into the above helper
```

> - The token audience matches the expected value passed into the above helper

Reviewer: ```suggestion
- that the token audience matches the expected value passed into the above helper
```

See the instructions for the [Okta JWT Verifier for Java](https://github.com/okta/okta-jwt-verifier-java).

### Required Libraries

> For validating a JWT, you will need the Okta JWT Verifier for Java library.

Reviewer: ```suggestion
To validate a JWT, you will need the Okta JWT Verifier for Java library.
```

> Include the below lines in your Maven pom.xml:

Reviewer: ```suggestion
Include the lines below in your Maven `pom.xml`:
```

```xml
<dependency>
   <groupId>com.okta.jwt</groupId>
   <artifactId>okta-jwt-verifier</artifactId>
   <version>${okta-jwt.version}</version>
</dependency>
  
<dependency>
  <groupId>com.okta.jwt</groupId>
  <artifactId>okta-jwt-verifier-impl</artifactId>
  <version>${okta-jwt.version}</version>
  <scope>runtime</scope>
</dependency>
```

### Validate Access Token

> Create the Okta JWT Verifier using `JwtVerifiers` class:

Reviewer: ```suggestion
Create the Okta JWT Verifier using the `JwtVerifiers` class:
```

```java
AccessTokenVerifier jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
    .setIssuer("https://{yourOktaDomain}/oauth2/default")
    .setAudience("api://default")                // defaults to 'api://default'
    .setConnectionTimeout(Duration.ofSeconds(1)) // defaults to 1s
    .setReadTimeout(Duration.ofSeconds(1))       // defaults to 1s
    .build();
```

> This helper class configures a JWT parser with the details found via the [OpenID Connect discovery endpoint](https://openid.net/specs/openid-connect-discovery-1_0.html).  The public keys used to validate the JWTs will also be retrieved and cached automatically.

Reviewer: ```suggestion
This helper class configures a JWT parser with the details found through the [OpenID Connect discovery endpoint](https://openid.net/specs/openid-connect-discovery-1_0.html).  The public keys used to validate the JWTs will also be retrieved and cached automatically.
```

Reviewer: Would it be more appropriate to link to the docs for our OpenID discovery endpoint? Or is this not using our endpoint?

Author: Not sure on this one. @bdemers your thoughts?

Reviewer 2: +1, this content is a little dated, so updating to point to Okta links would be better (and adds more cross-linking)

Author: @jakubvul-okta just to double check - https://developer.okta.com/docs/reference/api/oidc/#well-knownoauth-authorization-server is the right link to reference from here?

> After you have a `JwtVerifier` from above section and a `access_token` from a successful login, or from the `Bearer token` 

Reviewer: ```suggestion
After you have a `JwtVerifier` from the above section and an `access_token` from a successful sign in, or from a Bearer token 
```

> in the authorization header, you will need to make sure that this is still valid. All you need to do is call the 

Reviewer: ```suggestion
in the authorization header, you will need to make sure that it is still valid. All you need to do is call the 
```

`decode` method (where `jwtString` is your access token in string format).

```java
Jwt jwt = jwtVerifier.decode(jwtString);
```

This will validate your JWT for the following:

- Token expiration date

> - Valid token not before date

Reviewer: Not sure what this means.

Reviewer 2: 🤔 The "not before date" is the start date/time of when the token is valid, the expiration is the end. Maybe we could just say something more generic? Like "The token's date claims are valid" (🤮 but I hope you get the idea)

- The token issuer matches the expected value passed into the above helper
- The token audience matches the expected value passed into the above helper

> The result from the decode method is a `Jwt` object which you can introspect additional claims by calling:

Reviewer: ```suggestion
The result of the decode method is a `Jwt` object which you can introspect for additional claims by calling:
```
